metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/MetricDataAggregator.java

```diff
@@ -121,28 +155,30 @@ public void add(
         // This is ok.  Any new keys will be picked up on subsequent flushes.
         //TODO: use two maps and swap between, to ensure 'perfect' segmentation?
         List<MetricDatum> metricData = new ArrayList<>();
+        final Date now = new Date();
         for (DatumKey key : statisticsMap.keySet()) {
             StatisticSet value = statisticsMap.remove(key);
+            if (value == null) {
```

Comment:
Would it be better to filter when we add these to the map rather than storing them just to drop them later?

Reply:
The reason for the null check is because this is a concurrent map, which doesn't make guarantees between `keySet()` and `remove()`. Certainly, this is the only place in code where things are being removed, but I believe that this is the intended usage pattern of these two methods.